docs/kb/function-tutorial-deploy.md

```diff
@@ -0,0 +1,212 @@
+---
+layout: docs
+toc_id: function-tutorial-deploy
+title: VMware Event Broker Appliance - Function Deploy
+description: VMware Event Broker Appliance - Function Tutorial - Function Deploy
+permalink: /kb/function-tutorial-deploy
+cta:
+ title: What's next?
+ description: Continue devloping and deploying functions
+ actions:
+    - text: Go back to the Function Tutorial Intro [Function Tutorial - Function Intro](function-tutorial-intro)
+    - text: Go back to Modify and Test a Function Locally [Function Tutorial - Function Modify and Test](function-tutorial-modtest)
+    - text: Go to [Deploy Functions](use-functions).  This page also shows how to deploy a function using the vSphere UI instead of using kubectl.
+    - text: Look at the [Build Functions](contribute-functions) page for detailed instructions on how to build your own functions.  
+    - text: Look at [Troubleshoot Functions](troubleshoot-functions) on help to troubleshoot your functions.
+
+---
+
+# In-depth function tutorial - Deploy a Function to the VEBA Appliance
+
+This part of the tutorial will go over:
+- Pushing the [kn-ps-slack](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/master/examples/knative/powershell/kn-ps-slack) Docker container to the DockerHub registry.  For this example, we will use the modification we developed in the previous example and will restrict sending a Slack webhook when a Virtual Machine is powered off AND the VM Name starts with "prod".
+- Create a Kubernetes secret that contains the sensitive Slack webhook address.
+- Deploy the [kn-ps-slack](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/master/examples/knative/powershell/kn-ps-slack) function to the Kubernetes cluster on the VEBA appliance.
+- Verify function operation.
+
+## Table of Contents
+- [Push the Docker container to the DockerHub registry](#push-the-docker-container-to-the-dockerhub-registry)
+- [Introduction to the Kubernetes vmware-functions namespace](#introduction-to-the=kubernetes-vmware-functions-namespace)
+- [Deploy kn-ps-slack Function to the VEBA Appliance](#deploy-kn-ps-slack-function-to-the-veba-appliance)
+
+## Push the Docker container to the DockerHub registry
+Since we made a change to the code in the handler.ps1 file that we'd like to use, we will need to push our new Docker container to the DockerHub registry.  This will then make it available to be pulled down into VEBA's Kubernetes cluster.  The location of the container to use is referenced in the function.yaml file:
+
+<img src="./img/kn-ps-slack-function2.png" width="40%" align="center" class="border m-1 p-1"/>
+
+You can see that the default container referenced is: "us.gcr.io/daisy-284300/veba/kn-ps-slack:1.4".  We will replace this with our own container address.
+
+First, open a command prompt/terminal and push the docker image (replace docker-username):
```

Comment:
nit: inconsistent caps in docker (vs. Docker) - same for DockerHub

Reply:
Fixed.